internal/db/collection_define.go

```diff
@@ -148,12 +152,27 @@ func (db *DB) patchCollection(
 	if err != nil {
 		return err
 	}
+	newDefinitions := make([]client.CollectionDefinition, len(existingCols))
+	updatedColsByID := make(map[uint32]struct{})
+	for i, col := range existingCols {
+		newDefinitions[i].Schema = col.Schema()
+		newDefinitions[i].Description = newColsByID[col.ID()]
+		updatedColsByID[col.ID()] = struct{}{}
+	}
+	// append new cols
+	for id, col := range newColsByID {
+		if _, ok := updatedColsByID[id]; ok {
+			continue
+		}
+		newDefinitions = append(newDefinitions, client.CollectionDefinition{Description: col})
+	}
 
-	err = db.validateCollectionChanges(ctx, cols, newColsByID)
+	err = db.validateCollectionChanges(ctx, existingDefinitions, newDefinitions)
 	if err != nil {
 		return err
 	}
 
+	txn := mustGetContextTxn(ctx)
```

Comment:
question: I would have expected this like to be hit by test, not sure why it is not. Maybe due to ci failures?

Reply:
Ci issue